common/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/network/GenericNetworkQuery.kt

```diff
@@ -31,3 +31,49 @@ GetPasswordTokenInteractor) : PaginationOffsetLimitQuery(identifier ?: "$offset-
 class OAuthPasswordObjectQuery<T>(value: T, val getPasswordTokenInteractor: GetPasswordTokenInteractor) : ObjectQuery<T>(value),
     OAuthClientQuery
 
+abstract class GenericNetworkQuery(
+    val path: String,
+    val params: List<Pair<String, String>>?,
+    val headers: List<Pair<String, String>>?,
+    cacheKey: String? = null) : KeyQuery(cacheKey ?: path)
```

Comment:
it shouldn't be called **cacheKey** as it could be used for other things. I would just call it **key**

Reply:
The problem here is that **key** is already a **KeyQuery** property and it is not open, so we cannot override. Should we **open** this property?

Reply:
now it is called **key**